merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostProfileSegmentsAction.java

```diff
@@ -46,13 +47,32 @@ public <Dynamics> void apply(
       this.statement.setLong(2, profileRecord.id());
       setTimestamp(this.statement, 3, timestamp);
       setTimestamp(this.statement, 4, simulationStart);
-      this.statement.setString(5, serializeDynamics(dynamics, dynamicsP));
+      if (dynamics.isPresent()) {
+        this.statement.setString(5, serializeDynamics(dynamics.get(), dynamicsP));
+        this.statement.setBoolean(6, false);
+      } else {
+        this.statement.setString(5, "null");
```

Comment:
```sql
dynamics jsonb null
```
Given this definition of `dynamics` what's the semantic difference between a `null` `dynamics` and a `dynamics` with a JSON value of `null`?

Reply:
Its very blurry. I should have left that as `not null` though, I'll change it.

In the database, pg will convert JSON nulls into SQL nulls, but specifying `jsonb not null` doesn't restrict your ability to put JSON nulls into the column. So in effect, `jsonb not null` is a lie. If you say that the SQL value is `null`, that's an error. But if you say the value is a json object of `null`, it says "lol whatever you want boss" and promptly converts it into a SQL null without error. Its madness

Conceptually speaking from the user perspective, `null` is a legitimate value that they can use at their discretion. "Gaps" specifically mean unknown value, and I guess its more closely related to `void`.

Reply:
Ohh! Interesting. Thanks for the explanation, that's very good to know.